openpype/hosts/maya/plugins/publish/collect_pointcache.py

```diff
@@ -42,3 +42,21 @@ def process(self, instance):
         if proxy_set:
             instance.remove(proxy_set)
             instance.data["setMembers"].remove(proxy_set)
+
+        # Collect user defined attributes.
+        if not instance.data.get("includeUserDefinedAttributes", False):
+            return
+
+        all_nodes = (
+            instance.data["setMembers"] + instance.data.get("proxy", [])
+        )
```

Comment:
Isn't all nodes supposed to be the nodes in the instance, like `nodes = list(instance)`?

Set members is only the direct member of the set. It doesn't include e.g. children (and thus also not e.g. children shapes). And thus that's why you also need to explicitly get the shapes. If you use `list(instance)` you shouldn't need to.

Reply:
Yup good point.